detection_rules/devtools.py

```diff
@@ -721,21 +739,25 @@ def add_github_meta(this_rule: TOMLRule, status: str, original_rule_id: Optional
 
 @dev_group.command('deprecate-rule')
 @click.argument('rule-file', type=Path)
+@click.option('--deprecation-folder', '-d', type=Path)
 @click.pass_context
-def deprecate_rule(ctx: click.Context, rule_file: Path):
+def deprecate_rule(ctx: click.Context, rule_file: Path, deprecation_folder: Path):
     """Deprecate a rule."""
     version_info = loaded_version_lock.version_lock
     rule_collection = RuleCollection()
     contents = rule_collection.load_file(rule_file).contents
     rule = TOMLRule(path=rule_file, contents=contents)
 
-    if rule.contents.id not in version_info:
+    if rule.contents.id not in version_info and not RULES_CONFIG.bypass_version_lock:
         click.echo('Rule has not been version locked and so does not need to be deprecated. '
-                   'Delete the file or update the maturity to `development` instead')
+                   'Delete the file or update the maturity to `development` instead.')
         ctx.exit()
 
     today = time.strftime('%Y/%m/%d')
-    deprecated_path = rule.get_base_rule_dir() / '_deprecated' / rule_file.name
+    if not deprecation_folder:
+        deprecation_folder = RULES_CONFIG.rule_dirs[0] / '_deprecated'
```

Comment:
better to be explicit than implicit - with no guarantee of ordering, it seems somewhat random. This could even change internally, if say we added hunting to the 0 index.


```suggestion
```

There are a few alternative better options:

* you can use click callbacks (or default as a function) to make the default `rule.get_base_rule_dir() / '_deprecated' / rule_file.name` if no custom rules dir envvar is set otherwise `require`
* update the config to explicitly set a deprecated folder (with comments that its not needed in some instances)
* just make it an argument or required option always, where even we must explicitly call the deprecated dir

Reply:
This can be done in another PR along with the other examples `DEFAULT_PREBUILT_RULES_DIRS[0]`